test/unit/remote-config/remote-config-api-client.spec.ts

```diff
@@ -361,7 +480,7 @@ describe('RemoteConfigApiClient', () => {
     });
   }
 
-  function runErrorResponseTests(rcOperation: () => Promise<RemoteConfigTemplate>): void {
+  function runEtagHeaderTests(rcOperation: () => Promise<RemoteConfigTemplate>): void {
```

Comment:
Added this function to extract etag header validation tests from `runErrorResponseTests` function. That way I could re-use `runErrorResponseTests` for `list versions` operation. :)